ql/src/java/org/apache/hadoop/hive/ql/exec/repl/ReplDumpTask.java

```diff
@@ -376,13 +405,68 @@ private ReplicationSpec getNewEventOnlyReplicationSpec(Long eventId) {
     return rspec;
   }
 
-  Long bootStrapDump(Path dumpRoot, DumpMetaData dmd, Path cmRoot, Hive hiveDb) throws Exception {
+  private void dumpTableListToDumpLocation(List<String> tableList, Path dbRoot, String dbName, boolean isIncLoad,
+                                           HiveConf hiveConf) throws IOException, LoginException {
+    // Empty list will create an empty file to distinguish it from db level replication. If no file is there, that means
+    // db level replication. If empty file is there, means no table satisfies the policy.
+    if (tableList == null) {
+      LOG.debug("Table list file is not created for db level replication.");
+      return;
+    }
+
+    // For incremental dump the table list is dumped in _tables/dbname file and for bootstrap dump, its dumped in
+    // dbname/_tables file.
+    Path tableListFile;
+    if (isIncLoad) {
+      tableListFile = new Path(dbRoot, ReplUtils.REPL_TABLE_LIST_FILE_NAME);
+      tableListFile = new Path(tableListFile, dbName.toLowerCase());
+    } else {
+      tableListFile = new Path(dbRoot, dbName.toLowerCase());
+      tableListFile = new Path(tableListFile, ReplUtils.REPL_TABLE_LIST_FILE_NAME);
+    }
+
+    int count = 0;
+    while (count < FileUtils.MAX_IO_ERROR_RETRY) {
+      try (FSDataOutputStream writer = FileSystem.get(hiveConf).create(tableListFile)) {
```

Comment:
In retry mode, create might fail. Should we check if file exists and use open (overwrite mode) instead?

Reply:
create is by default overwrite mode ..if file is there ..it over writes